pkg/monitor/monitor/monitor_manager.go

```diff
@@ -613,3 +649,27 @@ func (m *MonitorManager) patchPVClaimRef(pv *corev1.PersistentVolume, patchPvcNa
 	}
 	return nil
 }
+
+func (m *MonitorManager) syncAssetSecret(monitor *v1alpha1.TidbMonitor, store *Store) error {
+	ns := monitor.Namespace
+	name := monitor.Name
+	tlsAssetsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            GetTlsAssetsSecretName(monitor.Name),
+			Namespace:       monitor.Namespace,
+			Labels:          buildTidbMonitorLabel(monitor.Name),
+			OwnerReferences: []metav1.OwnerReference{controller.GetTiDBMonitorOwnerRef(monitor)},
+		},
+		Data: make(map[string][]byte, len(store.TLSAssets)),
+	}
+	for key, asset := range store.TLSAssets {
```

Comment:
We should return early if `len(store.TLSAssets)==0`, the function `CreateOrUpdateSecret` will call API to kube-apiserver directly during each sync, which is not necessary if no cluster has TLS enabled.